CHANGELOG.md

```diff
@@ -37,6 +37,7 @@
   * `cortex_ingester_tsdb_block_postings_for_matchers_cache_requests_total`
   * `cortex_ingester_tsdb_block_postings_for_matchers_cache_skips_total`
 * [ENHANCEMENT] Compactor: Shuffle users' order in `BlocksCleaner`. Prevents bucket indexes from going an extended period without cleanup during compactor restarts. #10513
+* [ENHANCEMENT] Distributor, querier, ingester and store-gateway: Make use of LabelHints.Limit for label names and values requests. #10410
```

Comment:
I would phrase this in terms a user would understand: something like "add support for `limit` parameter for label names and values requests"